tests/_support/AcceptanceTester.php

```diff
@@ -256,6 +256,27 @@ public function iShouldSeeAnImageIsLinkedTo($image, $expectedUrl)
         $this->assertEquals($expectedUrl, $actualUrl);
     }
 
+    /**
+     * @When I click the table settings for :table
+     */
+    public function iClickTheTableSettings($table)
+    {
+        $this->click("#".$table);
+        $this->wait(1);
+    }
+
+    /**
+     * @Then I see checkbox :checkbox is checked
+     */
+    public function iSeeCheckboxIsChecked($checkbox)
```

Comment:
This function sounds very similar to another function already in this class: `iShouldSeeCheckboxIsChecked($checkbox)`. Since this function relates specifically to the `Table Settings` dialog box, I would probably rename it to something like `iShouldSeeCheckboxIsCheckedInTableSettingsDialogBox`.

Reply:
Hi @pli888, @rija,

The reason why I have created a new function because I wanted to test whether it could be checked or unchecked.  The way I wanted implement is to test checkbox could be checked/unchecked in one step. 
The implementation was a bit ambiguous,  and I should have reused the existing one.
 To make the scenarios more precise, I would update the scenarios in feature file as below:
```
  @wip @issue-917
  Scenario: Checkbox for files table in table settings exists
    When I am on "/dataset/100094"
    And I follow "Files"
    And I should see "Table Settings"
    And I click the table settings for "files_table_settings"
    Then I should see a check-box field "description"
    And I should see a check-box field "sample_id"
    And I should see a check-box field "type_id"
    And I should see a check-box field "format_id"
    And I should see a check-box field "size"
    And I should see a check-box field "date_stamp"
    And I should see a check-box field "location"
    And I should see a check-box field "attribute"

  @wip @issue-917
  Scenario:  Checkbox for files table in table settings can be checked
    When I am on "/dataset/100094"
    And I follow "Files"
    And I click the table settings for "files_table_settings"
    Then I check "description" checkbox
    And I should see "description" checkbox is checked
    And I check "sample_id" checkbox
    And I should see "sample_id" checkbox is checked
    And I check "type_id" checkbox
    And I should see "type_id" checkbox is checked
    And I check "sample_id" checkbox
    And I should see "sample_id" checkbox is checked
    And I check "format_id" checkbox
    And I should see "format_id" checkbox is checked
    And I check "size" checkbox
    And I should see "size" checkbox is checked
    And I check "date_stamp" checkbox
    And I should see "date_stamp" checkbox is checked
    And I check "location" checkbox
    And I should see "location" checkbox is checked
    And I check "attribute" checkbox
    And I should see "attribute" checkbox is checked

  @wip @issue-917
  Scenario:  Checkbox for files table in table settings can be unchecked
    When I am on "/dataset/100094"
    And I follow "Files"
    And I click the table settings for "files_table_settings"
    Then I uncheck "description" checkbox
    And I should see "description" checkbox is not checked
    And I uncheck "sample_id" checkbox
    And I should see "sample_id" checkbox is not checked
    And I uncheck "type_id" checkbox
    And I should see "type_id" checkbox is not checked
    And I uncheck "sample_id" checkbox
    And I should see "sample_id" checkbox is not checked
    And I uncheck "format_id" checkbox
    And I should see "format_id" checkbox is not checked
    And I uncheck "size" checkbox
    And I should see "size" checkbox is not checked
    And I uncheck "date_stamp" checkbox
    And I should see "date_stamp" checkbox is not checked
    And I uncheck "location" checkbox
    And I should see "location" checkbox is not checked
    And I uncheck "attribute" checkbox
    And I should see "attribute" checkbox is not checked
``` 
And the function `public function iSeeCheckboxIsChecked($checkbox)` will be removed which seems to be redundant.